bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/SwitchImpl.java

```diff
@@ -0,0 +1,67 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2023 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.forms.core.components.internal.models.v1.form;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.forms.core.components.internal.form.FormConstants;
+import com.adobe.cq.forms.core.components.models.form.Switch;
+import com.adobe.cq.forms.core.components.util.AbstractOptionsFieldImpl;
+
+@Model(
+    adaptables = { SlingHttpServletRequest.class, Resource.class },
+    adapters = { Switch.class,
+        ComponentExporter.class },
+    resourceType = { FormConstants.RT_FD_FORM_SWITCH_V1 })
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class SwitchImpl extends AbstractOptionsFieldImpl implements Switch {
+
+    // when a checkbox is not checked, it will still have a value representing its unchecked state, rather than having
+    // a null value. This configuration allows for a distinct value when the checkbox is in an unchecked state.
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL, name = "preserveUncheckedStateValue")
+    private Boolean preserveUncheckedStateValue;
+
+    @PostConstruct
+    private void initSwitchModel() {
```

Comment:
Looks like this is borrowed from checkbox, can you create an abstract check box implementation and extend switch from checkbox to re-use the same code, the same logic should exist for checkbox too

Reply:
When it comes to checkbox, we exclusively utilize enums and store them in two variables, namely 'checked' and 'unchecked.' However, in the case of switch, we store the enum values in an array while also saving the enumNames in a separate array. This is why we need to implement an override PostConstruct method